### PR TITLE
Define obt_env JS placeholder and $obt_env SCSS variable, and possibility to define custom vars

### PR DESCRIPTION
**Pull request by roland-vachter**
Define obt_env JS placeholder and $obt_env SCSS variable with the vaue of env in order to be able to make decisions based on the build type.

Also, by adding vars object, obt_env_{varKey} and $obt_env_{varKey} will be passed to the JS and SCSS resources.

### CODE REVIEW

**Comment by roland-vachter** [comment]
Custom variables are useful for build-time generated fingerprints (which then can be passed to o-assets).

File: lib/tasks/build.js [hunk 1]
```diff
@@ -87,6 +89,17 @@ module.exports.js = function(gulp, config) {
 				sourceMap: useSourceMaps
 			}));
 		}
+
+		plugins.push(new webpack.webpack.DefinePlugin({
+			'obt_env': `"${config.env}"`
```
Comment (AlbertoElias) [inline lib/tasks/build.js]: why not create a single object and then push it?

File: lib/tasks/build.js [hunk 2]
```diff
@@ -172,6 +185,15 @@ module.exports.sass = function(gulp, config) {
 
 		const combinedStream = combine.obj(
 			gulp.src(src),
+			inject.prepend(`$obt_env: '${config.env}';`),
```
Comment (AlbertoElias) [inline lib/tasks/build.js]: obt comes with a custom gulp plugin called prefixer, you could use that one instead of bringing in a new dependency

**Comment by roland-vachter** [comment]
I'm not actively using this feature anymore, so if you still find it useful, then fine, but I'm not relying on them anymore.

**Comment by AlbertoElias** [comment]
Left a couple of comments. If you could look into those and add tests for it, it'll be good to go

**Comment by AlbertoElias** [comment]
Ok, will close the pr then

**Pull request closed without merging**